pkg/crds/crds.go

```diff
@@ -0,0 +1,87 @@
+package crds
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var cephResources = []string{
+	"cephblockpoolradosnamespaces",
+	"cephblockpools",
+	"cephbucketnotifications",
+	"cephbuckettopics",
+	"cephclients",
+	"cephclusters",
+	"cephcosidrivers",
+	"cephfilesystemmirrors",
+	"cephfilesystems",
+	"cephfilesystemsubvolumegroup",
+	"cephnfses",
+	"cephobjectrealms",
+	"cephobjectstores",
+	"cephobjectstoreusers",
+	"cephobjectzonegroups",
+	"cephobjectzones",
+	"cephrbdmirrors",
+}
+
+const (
+	cephRookIoGroup          = "ceph.rook.io"
+	cephRookResourcesVersion = "v1"
+)
+
+var (
+	errorResourceNotFound      = errors.New("the server could not find the requested resource")
+	confirmationDeleteCRDPatch = map[string]interface{}{
+		"spec": map[string]interface{}{
+			"cleanupPolicy": map[string]string{
+				"confirmation": "yes-really-destroy-data",
+			},
+		},
+	}
+)
+
+func DeleteCustomResources(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace string) error {
+
+	for _, resource := range cephResources {
+		logging.Info(fmt.Sprintf("Getting ceph-rook resources kind %s", resource))
+		items, err := clientsets.GetResourcesDynamically(ctx, cephRookIoGroup, cephRookResourcesVersion, resource, clusterNamespace)
+		if err != nil {
+			if errors.As(err, &errorResourceNotFound) {
+				logging.Warning(err.Error())
+				continue
+			}
+			return err
+		}
+
+		for _, item := range items {
+			logging.Info(fmt.Sprintf("removing ceph-rook resource %s item %s", resource, item.GetName()))
```

Comment:
Wherever you log "ceph-rook", I'd suggest no need for it
```suggestion
			logging.Info(fmt.Sprintf("removing resource %s: %s", resource, item.GetName()))
```